modules/auxiliary/scanner/winrm/winrm_login.rb

```diff
@@ -72,6 +73,7 @@ def run_host(ip)
           timeout: 20,
           framework: framework,
           framework_module: self,
+          cache_file: datastore['WinrmKrb5Ccname'].blank? ? nil : datastore['WinrmKrb5Ccname'],
```

Comment:
I had an issue with this module when using Kerberos authentication with a CCache file and without `PASSWORD`. Apparently, the scanner validation requires a password, even if it is not necessary:
```
msf6 auxiliary(scanner/winrm/winrm_login) > run verbose=true RHOSTS=10.0.0.4 WinrmAuth=kerberos DOMAIN=mylab.local USERNAME=Administrator WinrmRhostname=dc01.mylab.local DomainControllerRhost=10.0.0.4 WinrmKrb5Ccname=/tmp/Administrator.ccache

[*] Error: 10.0.0.4: Metasploit::Framework::LoginScanner::Invalid Cred details can't be blank, Cred details can't be blank (Metasploit::Framework::LoginScanner::WinRM)
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

Reply:
So this is a pre-existing issue. I looked into fixing it, and I think we'd need to update how the credentials are iterated over since we're now supporting the Kerberos scenario in which the username is specified by the password is not. That's a bit more involved and I think it might be better to address it in a separate PR.

Reply:
I'm also happy to keep this out of scope from this PR; I had this as an edgecase in the Kerberos workflow doc - which suggested adding an extra datastore to track the scenario of the user not having creds available :+1: